SPECS-CROSS/cross-gcc/cross-gcc.spec

```diff
@@ -0,0 +1,161 @@
+%global debug_package %{nil}
+%global __os_install_post %{nil}
+
+# Globals which should be in a macro file.
+# These should be set programatically in the future.
+%global _host_arch      x86_64
+%global _target_arch    aarch64
+
+%global _tuple          %{_target_arch}-%{_vendor}-linux-gnu
+%global _cross_name     %{_target_arch}-%{_vendor}-linux-gnu
+
+# Folders which should be in our macro file
+%global _opt                /opt/
+%global _crossdir           /opt/cross/
+
+# Generally we include '/usr' in most paths.
+# Can we also use '/usr' for our paths? This will bring us in line with the
+# %%configure macro which sets these.
+%global _bindir            /bin
+%global _sbindir           /sbin
+%global _libdir            /lib
+%global _lib64dir          /lib64
+%global _libexecdir        /libexec
+%global _datadir           /share
+%global _docdir            /share/doc
+%global _includedir        /include
+%global _infodir           /share/info
+%global _mandir            /share/man
+%global _oldincludedir     /include
+
+# If we want our cross compile aware packges to also support native, we
+# need logic to switch modes something like this:
+%if "%{_target_arch}" != "%{_host_arch}"
+%global _cross_prefix       %{_crossdir}%{_tuple}/
+%global _cross_sysroot      %{_crossdir}%{_tuple}/sysroot/
+%global _cross_includedir   /usr/%{_host}/%{_tuple}/include/
+%global _cross_infodir      %{_crossdir}%{_tuple}/share/info
+%global _cross_bindir       %{_tuple}/bin
+%global _cross_libdir       %{_tuple}/lib
+%global _tuple_name         %{_tuple}-
+
+%global __strip %{_cross_prefix}%{_bindir}/%{_tuple_name}strip
+%global __objdump %{_cross_prefix}%{_bindir}/%{_tuple_name}objdump
+%else
+%global _cross_prefix       %{nil}
+%global _cross_sysroot      %{nil}
+%global _cross_includedir   %{_includedir}
+%global _cross_infodir      %{_infodir}
+%global _cross_bindir       %{_bindir}
+%global _cross_libdir       %{_libdir}
+%global _tuple_name         %{nil}
+%endif
+
+Summary:        Contains the GNU compiler collection, binutils, glibc, kernel headers
+Name:           %{_cross_name}-cross-gcc
+Version:        0.1.0
+Release:        1%{?dist}
+License:        GPLv2+
+Vendor:         Microsoft Corporation
+Distribution:   Mariner
+Group:          Development/Tools
+BuildRequires:  %{_cross_name}-binutils
+BuildRequires:  %{_cross_name}-kernel-headers
+BuildRequires:  %{_cross_name}-glibc-bootstrap2
+BuildRequires:  %{_cross_name}-gcc-bootstrap3
+BuildRequires:  grep
+AutoReqProv:    no
+ExclusiveArch:  x86_64
+Conflicts:      %{_cross_name}-gcc-bootstrap
+Conflicts:      %{_cross_name}-gcc-bootstrap2
+Conflicts:      %{_cross_name}-gcc-bootstrap3
+Conflicts:      %{_cross_name}-glibc-bootstrap
+Conflicts:      %{_cross_name}-glibc-bootstrap2
```

Comment:
Why do we need so many bootstrap packages? Btw. for the sake of building Mariner on top of Ubuntu we've been building our bootstrap packages inside `toolkit/scripts/toolchain/container/toolchain_build_in_chroot.sh`. I like relying solely on specs, but I think we shouldn't multiply these bootstrap packages, as they are hard to maintain. See: recent issue with `cyrus-sasl-bootstrap` and `openldap`. This might be a bigger discussion, but my initial 3 ideas for this are:
- Duplicate the approach from `toolkit/scripts/toolchain/container/toolchain_build_in_chroot.sh`, or
- use `Obsoletes` instead of `Conflicts` similar to what `cyrus-sasl` is doing with `cyrus-sasl-bootstrap` or `systemd` with `systemd-bootstrap`, or
- **[bigger item]** extend our toolkit to support bootstrap packages built from one spec through the use of an additional `bootstrap` macro - Fedora already does something similar in some of their packages, I believe.